proposed/phpdoc-tags.md

```diff
@@ -644,13 +644,9 @@ RECOMMENDED in case of complicated return structures, such as associative arrays
 The @return tag MAY have a multi-line description and does not need explicit
 delimiting.
 
-It is RECOMMENDED to use this tag with every function and method. An exception to
-this recommendation, as defined by the Coding Standard of any individual project,
-MAY be:
-
-   **functions and methods without a `return` value**: the @return tag MAY be
-   omitted here, in which case an interpreter MUST interpret this as if
-   `@return void` is provided.
+It is RECOMMENDED to use this tag with every function and method. The @return
+tag MAY be omitted by functions and methods without a `return` value, in which
+case an interpreter MUST interpret this as if `@return void` is provided.
```

Comment:
I don't agree with this (allowing tag to be emitted in the case of `void` return and then treating everything without it as returning `void`).

- As above, a user should not have to add a `@return int` to a function with an `int` return type. This rule would mean treating that function as if it had returned `void`. That makes no sense to me.
- Code with `return` statements but no `@return` is far more common, in my experience, than code with incorrect `@return` types. If implementers of static analysis tools abided by this rule we'd break a lot of code that uses third-party libraries:
    ```php
    $a = third_party_function_without_docblock(); // AssignmentToVoid issue in Psalm
    ```

I propose instead always requiring an `@return` type of some description if there's no typehint given.

Reply:
Yep, was going to point this out after Jan's comment reminded me of return type in code. This verbiage predates that being added to PHP.

Reply:
I'd suggest
```
It is RECOMMENDED to use this tag with every function and method.
If no `@return` type is given, and no return type declaration is provided,
an interpreter MUST interpret this as if `@return mixed` is provided.
```

Reply:
I don't think any implementation can be required to parse the method body itself in search of `return` statements, in order to derive types when they are not already specified by tag or code signature.  In those cases, @muglug 's suggestion of defaulting to `mixed` seems good enough to me.

Reply:
I just want to mention that `MUST` seems to _exclude_ option of using more precise type, while implementation should be allowed to do so.